src/components/Product-card/product-card.jsx

```diff
@@ -1,10 +1,96 @@
+import { useAuth } from "../../contexts/auth-context";
 import "../../pages/Product-listing/product-listing.css";
 
-const ProductCard = ({ productcard }) => {
-  const { title, price, rating, productImg } = productcard;
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import { useState } from "react";
+import { useWishlist } from "../../contexts/wishlist-context";
+
+const ProductCard = ({productcard}) => { //destructuring the product card object from props
+  const {state, dispatch} = useWishlist();
+
+   //extracting the wishlist from state where we're storing it
+
+   const {wishlist} = state;
+
+
+
+  const { title, price, rating, productImg } = productcard; // destrructuring all the keys from productcard
+
+
+  const {auth: {Authenticated, token}} = useAuth();
+
+  const navigate = useNavigate();
```

Comment:
Suggestion: You can initialize `navigate` in authContext and import it wherever needed.